### ISSUE TITLE
Bug: BatchNormalization causes Input layer to lose name

### ISSUE THREAD

**Issue by bvbellomo** [keras-team/tf-keras#506]
If I start a functional network with:
inputs = Input(shape=inputShape, name="data")
inputs = BatchNormalization()(inputs)
outputs =  Dense(inputs)

model.summary doesn't show my batch normalization and loses the name of my input layer:
input_1 (InputLayer)
dense (Dense)

If I omit my BatchNormalization:
inputs = Input(shape=inputShape, name="data")
#inputs = BatchNormalization()(inputs)
outputs =  Dense(inputs)

model.summary shows what I expect:
data (InputLayer)
dense (Dense)

A BatchNormalization anywhere except immediately after an input layer appears to work correctly
